react/components/GoogleMap.tsx

```diff
@@ -0,0 +1,376 @@
+
+      const differenceLng =
+        Math.abs(referenceCenterLng - currentCenter.lng()) * 1000
+
+      setShouldSearchArea(differenceLat > DISTANCE || differenceLng > DISTANCE)
+      setMapCenterLatLng(currentCenter)
+    }, [map, setMapCenterLatLng, setShouldSearchArea])
+
+    const mapMounted = (node: HTMLDivElement | null) => {
+      if (node == null || map !== null) {
+        return
+      }
+
+      const googleMap = new google.maps.Map(node, {
+        zoom: STANDARD_ZOOM,
+        disableDefaultUI: true,
+        mapTypeControl: false,
+        fullscreenControl: false,
+        streetViewControl: false,
+        clickableIcons: false,
+        styles: [
+          {
+            featureType: 'poi',
+            stylers: [{ visibility: 'off' }],
+          },
+          {
+            featureType: 'transit',
+            elementType: 'labels.icon',
+            stylers: [{ visibility: 'off' }],
+          },
+        ],
+        center:
+          center.length === 2 ? geoCoordinatesToLatLng(center) : undefined,
+      })
+
+      if (bounds) {
+        googleMap.fitBounds(bounds)
+      }
+
+      if (center.length === 2) {
+        toggleSearchArea()
+
+        if (isLargeScreen) {
+          googleMap.panBy(PAN_LEFT_LAT, PAN_LEFT_LNG)
+        }
+      }
+
+      setMap(googleMap)
+    }
+
+    const prevBoundsRef = useRef<google.maps.LatLngBounds | undefined>(
+      undefined
+    )
+
+    useEffect(() => {
+      if (
+        map == null ||
+        isGeoCoordinatesEqual(
+          [map.getCenter()?.lng(), map.getCenter()?.lat()],
+          center
+        )
+      ) {
+        return
+      }
+
+      if (center.length !== 2) {
+        return
+      }
+
+      map.setCenter(geoCoordinatesToLatLng(center))
+
+      if (map.getZoom() !== STANDARD_ZOOM) {
+        map.setZoom(STANDARD_ZOOM)
+      }
+
+      if (isLargeScreen) {
+        map.panBy(PAN_LEFT_LAT, PAN_LEFT_LNG)
+      }
+    }, [map, center, isLargeScreen])
+
+    useEffect(() => {
+      if (
+        map == null ||
+        !bounds ||
+        bounds.equals(prevBoundsRef.current ?? null)
+      ) {
+        return
+      }
+
+      map.fitBounds(bounds)
+
+      if ((map.getZoom() ?? STANDARD_ZOOM) > STANDARD_ZOOM) {
+        map.setZoom(STANDARD_ZOOM)
+      }
+
+      if (isLargeScreen) {
+        map.panBy(PAN_LEFT_LAT, PAN_LEFT_LNG)
+      }
+
+      prevBoundsRef.current = bounds
+    }, [map, bounds, isLargeScreen])
+
+    useMapEvent(map, 'dragend', () => {
+      toggleSearchArea()
+    })
+
+    useMapEvent(map, 'click', () => {
+      setActiveSidebarState(LIST)
+      onClick?.()
+    })
+
+    useMapEvent(map, 'zoom_changed', () => {
+      setActiveState(activeState)
+    })
+
+    useMapEvent(map, 'dragstart', () => {
+      setActiveState(activeState)
+    })
+
+    useEffect(() => {
+      if (map == null) {
+        return
+      }
+
+      // TODO: can we move this `querySelector` call to use React refs instead?
```

Comment:
It makes sense to me. What do you think about opening a new task and attaching it to our backlog?

Reply:
I worry that it won't ever be prioritized for us to fix, so it would be just a "dead" task :-/

Reply:
But it will be better tracked on the board than on this TODO. If you don't wanna open a new task, you can just delete this comment.

Reply:
Also, this is a good example of a `good first bug` task to put to someone do in the onboarding process.

Reply:
you convinced me 😄 created here: https://vtex-dev.atlassian.net/browse/CHK-1098